pkg/generate/app/cmd/newapp.go

```diff
@@ -992,7 +1000,13 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 	}
 
 	if len(c.Name) > 0 {
-		if err := c.validateEnforcedName(); err != nil {
+		if err := validateEnforcedName(c.Name); err != nil {
+			return nil, err
+		}
+	}
+
+	if len(c.To) > 0 {
+		if err := validateOutputImageReference(c.To); err != nil {
```

Comment:
As I commented above, it can be either a DockerImage or a ImageStreamTag, depending on the `--to-docker` flag.